### ISSUE TITLE
As an employee, I want to know how much I worked, by being able to get a summary of hours worked for each week.

### ISSUE THREAD

**Issue by ECourant** [ECourant/standards#3]
This should be handled by the following url `/api/shifts/mine/summary` which should return the following data grouped by the week number of the year in descending order.

- [x] A start date for that week
- [x] An end date for that week
- [x] Number of shifts for that week
- [x] Total hours worked (the total hours worked each day - the break)
- [x] Total hours on break

Issues:
- [x] If a shift goes into the next week, only count the hours in the current week.
- [x] If a shift goes into the next week, make sure that the hours are represented in that week.

I think this could be achieved with something like this:
date_trunc('week', start_time::date) AS week
```sql
SELECT
    employee_id,
    week,
    to_date(week::text, 'yyyyww') AS week_start_date, --I will need to find out the format the week number is returned in
    to_date(week::text, 'yyyyww') + '7 Days'::interval AS week_end_date,
    SUM(hours),
    SUM(breaks)
FROM (
    SELECT
        employee_id,
        date_trunc('week', start_time::date) AS week,
        ((end_time - start_time) - {{{something here to convert break to interval}}}) AS hours,
        break AS breaks
    FROM shifts
)
WHERE employee_id = ?
GROUP BY
    employee_id,
    week
ORDER BY
    week DESC
```

**Comment by ECourant** [comment]
I need to figure out how to handle shift's carrying into the next week.

**Comment by ECourant** [comment]
Fixed completely in 0b07236

**Comment by ECourant** [comment]
I NEED TO ADD CALCULATIONS FOR SHIFTS IN THE FUTURE OR IN PROGRESS.
If the current timestamp is only half way through an 8 hour shift, that shift should show as 4 hours worked. 

and shifts in the future completely should just show 0 hours worked.

With that being said, i should add fields for hours scheduled vs hours worked.

**Comment by ECourant** [comment]
I've fixed this in commit 0ae3d88
